src/deque.jl

```diff
@@ -38,6 +38,8 @@ isrear(blk::DequeBlock) =  blk.next === blk
 # reset the block to empty, and position
 
 function reset!(blk::DequeBlock{T}, front::Int) where T
+    empty!(blk.data)
+    blk.data = Vector{T}(undef, blk.capa)
```

Comment:
Why empty it if its just going to be over written with a newly allocated array?
I would think just:
```suggestion
    empty!(blk.data)
```
would be enough.
(and it's nonallocating)

Reply:
The underlying the assumption of a block is that it's data field will have already `data` field allocated with `capa` number of elements. Just emptying and not allocating block's `data` leads to the problem that once you have cleared the deque, you can't push elements into it again. That's why I've added new testset `push! after empty!` to check this scenario.

Reply:
So we want:
```suggestion
    # Fill it with undefs back to capacity
    empty!(blk.data); resize!(blk.data, blk.capa)
```
?

Reply:
yes. made this change